### ISSUE TITLE
`String.Format` and params array

### ISSUE THREAD

**Issue by scottdorman** [dotnet/runtime#42923]
The `String.Format` methods take a `params` array as an argument which looks like it's been enabled for nullable reference types.

https://github.com/dotnet/runtime/blob/118eee973af81119fd00fe75bc9c14ce3192ff1e/src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs#L481-L491

However, in the [docs](https://docs.microsoft.com/en-us/dotnet/api/system.string.format?view=netcore-3.1#System_String_Format_System_String_System_Object___) the signature is shown as 

`public static string Format (string format, params object[]? args);`

Is this a docs bug or should the implementation be `params object[]? args`?

Isn't one ( `params object[]? args`) effectively "a parameters array of object, where the entire argument could be `null`) while the other ( `params object?[] args`) effectively "a parameters array of nullable object, where the argument itself isn't expected to be `null`"?

**Comment by stephentoub** [comment]
> Is this a docs bug

Yup. The source has the correct signature, as does the reference assembly and resulting IntelliSense in Visual Studio:
![image](https://user-images.githubusercontent.com/2642209/94745634-15501d80-0349-11eb-9aed-11058972ebee.png)

As you say, `object[]?` is a nullable array of non-nullable objects whereas `object?[]` is a non-nullable array of nullable objects (and for completeness, `object?[]?` is a nullable array of nullable objects).  string.Format takes a non-nullable array of nullable objects, `object?[]`.

**Comment by stephentoub** [comment]
@carlossanlop, @terrajobst, what's the right way to handle this?  I'm assuming this is actually a bug in whatever tooling is used to generate the signature portion of the docs?

**Comment by scottdorman** [comment]
I can edit the docs page and submit a PR, but that doesn't sound like the right approach here and my change would get overwritten by the tooling whenever it's generated again.

**Comment by scottdorman** [comment]
> `object?[]?` is a nullable array of nullable objects)

That makes my head hurt! LOL

**Comment by carlossanlop** [comment]
> _I can edit the docs page and submit a PR, but that doesn't sound like the right approach here and my change would get overwritten by the tooling whenever it's generated again._

Thanks for offering, @scottdorman, but we need the xml to get updated via our tooling.

@joelmartinez @TianqiZhang @mimisasouvanh can you please take a look? The mdoc process placed the `?` in the wrong position.

This is how it's defined in the ref file: 
`System.String Format(System.String format, params object?[] args)`

https://github.com/dotnet/runtime/blob/1bc493ab8b65d167180ad1f41f3a6292d28d7094/src/libraries/System.Runtime/ref/System.Runtime.cs#L3516

I'll transfer this issue to the dotnet-api-docs repo.
